File: getprime.py
#!/usr/bin/env python

from sys import argv
from time import time


def prime(num):
    if num < 2:
        return False
    if num == 2:
        return True
    if num % 2 == 0:
        return False

    for i in range(3, num//2, 2):
        if num % i == 0:
            return False

    return True


def yieldprime(target):
    count = 1
    primes = 2
    while True:
        if count == target:
            return primes
        primes += 1
        if prime(primes):
            count += 1


if __name__ == '__main__':
    if len(argv) > 1:
        print(yieldprime(int(argv[1])))

File: getprime.js
#!/usr/bin/env node

exports.prime = prime
exports.yieldprime = yieldprime


function prime(num) {
    if (num < 2)
        return false
    if (num == 2)
        return true
    if (num%2 == 0)
        return false

    for (let i = 3; i < num/2; i += 2) {
        if (num%i == 0)
            return false
    }

    return true
}


function yieldprime(target) {
    let count = 1,
        primes = 2
    while (true) {
        if (count == target)
            return primes
        primes++
        if (prime(primes))
            count++
    }
}


if (!module.parent)
    if (process.argv.length > 2)
        console.log(yieldprime(parseInt(process.argv[2])))
